Resources/public/js/views/services/ez-locationviewviewservice.js

```diff
@@ -208,6 +209,42 @@ YUI.add('ez-locationviewviewservice', function (Y) {
         },
 
         /**
+         * translate event handler, makes the application to navigate to edit content available
+         * in the facade with given language and base language
+         *
+         * @method _translateContent
+         * @protected
+         * @param {EventFacade} e
+         * @param {String} e.languageCode language code of new translation
+         * @param {Boolean} e.isBaseLanguage defines if new translation is basing on another one or not
+         *                  if it's set to true then languageCode of base language will be set, otherwise
+         *                  base language will be ommited
+         * @param {String} e.baseLanguageCode language code of existing translation on which 
+         *                  new translation will be basing, it it's not set then current
+         *                  languageCode will be used
+         */
+        _translateContent: function (e) {
+            var app = this.get('app'),
+                baseLanguageCode = this.get('languageCode'),
+                routeParams = {
+                        id: e.content.get('id'),
+                        languageCode: e.languageCode,
+                    };
+
+            if (e.baseLanguageCode) {
```

Comment:
> 3 Here I have prepared the `translate` event: https://github.com/ezsystems/PlatformUIBundle/pull/280/files?diff=unified#diff-9b8359562331d6a0f062d74df2f6525bR226
It's called here: https://github.com/ezsystems/PlatformUIBundle/pull/280/files?diff=unified#diff-659f9de6ec2e07d2db4ea7912dc5e01cR151
It's bit complicated but I don't know if it could be simplier. Or maybe my approach is incorrect.

indeed that's too complicated because if the user did not choose a base language code you don't have to use the current language as the base. Basically this can be simplified to:
```js
            var routeParams = {
                    id: e.content.get('id'),
                    languageCode: e.languageCode,
                };

            if (e.baseLanguageCode) {
                routeParams.baseLanguageCode = e.baseLanguageCode;
            }
            this.get('app').navigate(
                app.routeUri('editContent', routeParams)
            );
```